clang/lib/Analysis/FlowSensitive/DataflowEnvironment.cpp

```diff
@@ -350,6 +350,13 @@ class ResultObjectVisitor : public RecursiveASTVisitor<ResultObjectVisitor> {
     return RecursiveASTVisitor<ResultObjectVisitor>::TraverseDecl(D);
   }
 
+  bool TraverseDecltypeTypeLoc(DecltypeTypeLoc Node) {
```

Comment:
There are other instances in the language that are unevaluated including: `typeid`, `noexcept`, `sizeof`. Do you anticipate similar problems with those?

Reply:
> There are other instances in the language that are unevaluated including: `typeid`, `noexcept`, `sizeof`. Do you anticipate similar problems with those?

Excellent point, thanks. I'll expand this PR to more generally exclude all unevaluated contexts. Will ping this PR when I've changed this.

Reply:
> It also turns out that, even though the operand of the noexcept operator in an unevaluated operand, it still shows up in the CFG for some reason

This surprises me, I can't think of a reason why we would want to have it in the CFG. This keyword appeards a lot in generic code, so maybe excluding it from the CFG might have measurable perf/memory benefits. I'd think it might be worth to give this a try and see if something breaks (in a separate PR).

Reply:
Makes sense, I'll take a look at that!